### PR TITLE
Fix all flake8 issues on python 3

### PR DESCRIPTION
**Pull request by quantum5**
Fix all flake8 issues uncovered by #8878 except the one in #9200.

### CODE REVIEW

File: emar.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def run():
           newargs[j] = full_newname
           to_delete.append(full_newname)
           contents.add(newname)
-        except:
+        except IOError:
```
Comment (sbc100) [inline emar.py]: I'm not sure exactly why the original author thought it was OK to fail here... so I don't know if IOError is enough to capture their intention.    Shame we dont have a use case or test case or a better comment.

File: emcc.py [hunk 2]
```diff
@@ -503,7 +503,7 @@ def run(args):
   if '--version' in args:
     try:
       revision = run_process(['git', 'show'], stdout=PIPE, stderr=PIPE, cwd=shared.path_from_root()).stdout.splitlines()[0]
-    except:
+    except CalledProcessError:
```
Comment (sbc100) [inline emcc.py]: Is this really the only thing that can go wrong here?  What if git is not found isn't that a differnt error type?

File: tests/poppler/poppler/gen-unicode-tables.py [hunk 3]
```diff
@@ -11,6 +11,11 @@
 HANGUL_S_BASE = 0xAC00
 HANGUL_S_COUNT = 19 * 21 * 28
 
+try:
+    unichr
```
Comment (kripken) [inline tests/poppler/poppler/gen-unicode-tables.py]: indentation looks wrong here, and below

File: emcc.py [hunk 4]
```diff
@@ -862,7 +862,7 @@ def optimizing(opts):
           newargs[i] = newargs[i + 1] = ''
           if key == 'WASM_BACKEND=1':
             exit_with_error('do not set -s WASM_BACKEND, instead set EMCC_WASM_BACKEND=1 in the environment')
-    newargs = [arg for arg in newargs if arg is not '']
+    newargs = [arg for arg in newargs if arg != '']
```
Comment (sbc100) [inline emcc.py]: hmm.. I wonder why python2 flake8 didn't catch this one?

File: emcc.py [hunk 5]
```diff
@@ -3528,7 +3528,7 @@ def parse_string_list(text):
     # simpler syntaxes
     try:
       return json.loads(text)
-    except:
+    except ValueError:
```
Comment (sbc100) [inline emcc.py]: Again, how can we be sure this is the only type of exception that can arise here?

File: emrun.py [hunk 6]
```diff
@@ -366,10 +366,10 @@ def kill_browser_process():
       else:
         try:
           subprocess.call(['pkill', processname_killed_atexit])
-        except OSError as e:
```
Comment (sbc100) [inline emrun.py]: Why didn't python2 flake8 already get these?

**Comment by quantum5** [comment]
When used together with #8878, this reduces the list of python 3 flake8 errors. With #9200, there should be no errors left.

File: tests/poppler/poppler/gen-unicode-tables.py [hunk 7]
```diff
@@ -11,6 +11,11 @@
 HANGUL_S_BASE = 0xAC00
```
Comment (sbc100) [inline tests/poppler/poppler/gen-unicode-tables.py]: This code is not written by us?  If its upstream code then better to at it to the exclude list.

File: tests/test_core.py [hunk 8]
```diff
@@ -7043,7 +7043,7 @@ def encode_utf8(data):
         for i in range(len(data)):
           data[i] = encode_utf8(data[i])
         return data
-      elif isinstance(data, unicode):
+      elif isinstance(data, type(u'')):
```
Comment (sbc100) [inline tests/test_core.py]: This seems strictly less readable to me.

File: tests/test_other.py [hunk 9]
```diff
@@ -4151,13 +4151,13 @@ def test_returncode(self):
     for code in [0, 123]:
       for no_exit in [0, 1]:
         for call_exit in [0, 1]:
-          for async in [0, 1]:
-            run_process([PYTHON, EMCC, 'src.cpp', '-DCODE=%d' % code, '-s', 'EXIT_RUNTIME=%d' % (1 - no_exit), '-DCALL_EXIT=%d' % call_exit, '-s', 'WASM_ASYNC_COMPILATION=%d' % async])
+          for async_ in [0, 1]:
```
Comment (sbc100) [inline tests/test_other.py]: Maybe rename to async_compile?

File: emcc.py (hunk 5 above)
Comment (quantum5) [reply emcc.py]: From [documentation](https://docs.python.org/3/library/json.html#json.loads):

> If the data being deserialized is not a valid JSON document, a `JSONDecodeError` [subclass of `ValueError`] will be raised.

This is the only kind of error we want to recover from.

File: emrun.py (hunk 6 above)
Comment (quantum5) [reply emrun.py]: I think our flake8 version was too old.

File: tests/test_core.py (hunk 8 above)
Comment (quantum5) [reply tests/test_core.py]: Would you prefer:

```python
try:
  unicode
except NameError:
  unicode = str
```

File: emcc.py (hunk 2 above)
Comment (quantum5) [reply emcc.py]: No, it's not. I already fixed this after the CI failed.

File: emcc.py (hunk 4 above)
Comment (quantum5) [reply emcc.py]: The version is too old.

File: tests/test_core.py (hunk 8 above)
Comment (sbc100) [reply tests/test_core.py]: Not really.  Just noticing out loud that this compatibility can make the code harder to read.

How would you do this in python3-only code?

Comment (quantum5) [reply tests/test_core.py]: I would have written `isinstance(data, str)` here. But perhaps a better way is doing `isinstance(data, bytes)` so `bytes` gets returned raw, and everything goes through `str` and then encoded.

File: tests/test_other.py (hunk 9 above)
Comment (quantum5) [reply tests/test_other.py]: Done.

File: tests/poppler/poppler/gen-unicode-tables.py (hunk 7 above)
Comment (quantum5) [reply tests/poppler/poppler/gen-unicode-tables.py]: Done.

File: emar.py (hunk 1 above)
Comment (quantum5) [reply emar.py]: Changed to `except Exception:`

**Comment by quantum5** [comment]
Checking this on CircleCI is the intention of #8878.

**Comment by kripken** [comment]
@quantum5 looks like this caused breakage on MacOS, see

https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket.appspot.com/8905225763363687616/+/steps/Emscripten_testsuite__upstream__other_/0/stdout

Looks like the change to `  except OSError:` isn't enough, in `macos_find_native_sdk_path`.

**Pull request merged**